packages/devtools_app/lib/src/screens/memory/panes/chart/controller/chart_pane_controller.dart

```diff
@@ -95,19 +100,34 @@ class MemoryChartPaneController extends DisposableController
   ValueListenable<bool> get paused => _paused;
   final _paused = ValueNotifier<bool>(true);
   void pause() => _paused.value = true;
-  Future<void> resume() async {
-    if (!_paused.value) return;
-    if (mode != DevToolsMode.connected) throw StateError('Not connected.');
-    await _chartConnection!.maybeConnect();
+  void resume() {
+    assert(mode == DevToolsMode.connected && _chartConnection != null);
     _paused.value = false;
   }
 
+  /// Connects when chart is first time expanded.
+  ///
+  /// If chart is already connected, does nothing.
+  /// Return true if connection was setup.
+  Future<bool> maybeConnect() async {
+    if (!_paused.value) return false;
+    if (mode != DevToolsMode.connected) return false;
+    await _chartConnection!.maybeInitialize();
+    return true;
+  }
+
   final isAndroidChartVisible = ValueNotifier<bool>(false);
+  void _calculateAndroidChartVisibility() {
+    data.isDeviceAndroid ??= _chartConnection!.isDeviceAndroid;
+    isAndroidChartVisible.value = data.isDeviceAndroid! &&
+        preferences.memory.androidCollectionEnabled.value;
+  }
 
-  void _updateAndroidChartVisibility() {
-    final isAndroid = data.isDeviceAndroid ?? _chartConnection!.isDeviceAndroid;
-    isAndroidChartVisible.value =
-        isAndroid && preferences.memory.androidCollectionEnabled.value;
+  ValueListenable<bool> isChartVisible = preferences.memory.showChart;
```

Comment:
Should this be grouped with the other variables? Also, could this be made into a getter instead?